File: 10F/teman-dekat.py
str_in = input().split(' ')

n = int(str_in[0])
d = int(str_in[1])

def f(xi, yi, xj, yj):
    return abs(xj-xi)**d + abs(yj-yi)**d

arr = []

for i in range(n):
    arr_in = input().split(' ')
    x = int(arr_in[0])
    y = int(arr_in[1])

    arr.append((x,y))

i=0
j=1

ans=[]

while i < len(arr) - 1:
    ans.append(f(arr[i][0], arr[i][1], arr[j][0], arr[j][1]))

    if j < len(arr) - 1:
        j += 1
    else:
        i += 1
        j = i + 1

print(min(ans), max(ans))

File: 10F/teman-dekat.go
package main

import (
	"fmt"
	"math"
)

var n, x, y, min, max int64
var d float64

func f(xi, xj, yi, yj int64) int64 {
	a := math.Abs(float64(xj - xi))
	b := math.Abs(float64(yj - yi))
	return int64(math.Pow(a, d) + math.Pow(b, d))
}

func main() {
	fmt.Scan(&n, &d)
	var arr = make([][]int64, int(n))
	for i := 0; i < int(n); i++ {
		arr[i] = make([]int64, 2)
		fmt.Scan(&x, &y)
		arr[i][0] = x
		arr[i][1] = y
	}
	i := 0
	j := 1
	var data = []int64{}
	for i < len(arr)-1 {
		data = append(data, f(arr[i][0], arr[j][0], arr[i][1], arr[j][1]))

		if j < len(arr)-1 {
			j += 1
		} else {
			i += 1
			j = i + 1
		}

	}

	for id, item := range data {
		if id == 0 {
			min = item
			max = item
		}
		if item > max {
			max = item
		}
		if item < min {
			min = item
		}
	}
	fmt.Print(min, max)
}
